File: VB/WpfApplication48/VerticalLineHelper.vb
Imports DevExpress.Xpf.Grid
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows
Imports System.Windows.Data
Imports System.Windows.Media

Namespace WpfApplication48
	Public Class VerticalLineHelper
		Public Shared Function GetLineThickness(ByVal obj As DependencyObject) As Double
			Return DirectCast(obj.GetValue(LineThicknessProperty), Double)
		End Function
		Public Shared Sub SetLineThickness(ByVal obj As DependencyObject, ByVal value As Double)
			obj.SetValue(LineThicknessProperty, value)
		End Sub
		Public Shared Function GetLineColor(ByVal obj As DependencyObject) As Brush
			Return DirectCast(obj.GetValue(LineColorProperty), Brush)
		End Function
		Public Shared Sub SetLineColor(ByVal obj As DependencyObject, ByVal value As Brush)
			obj.SetValue(LineColorProperty, value)
		End Sub
		Public Shared ReadOnly LineThicknessProperty As DependencyProperty = DependencyProperty.RegisterAttached("LineThickness", GetType(Double), GetType(VerticalLineHelper), New PropertyMetadata(0.0))
		Public Shared ReadOnly LineColorProperty As DependencyProperty = DependencyProperty.RegisterAttached("LineColor", GetType(Brush), GetType(VerticalLineHelper), New PropertyMetadata(Brushes.Black))
	End Class

	Public Class VerticalLineThicknessConverter
		Implements IValueConverter

		Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
			Return New Thickness(0, 0, DirectCast(value, Double), 0)
		End Function

		Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
			Throw New NotImplementedException()
		End Function
	End Class

	Public Class VerticalLineVisibilityConverter
		Implements IValueConverter

		Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
			Return (value IsNot Nothing) AndAlso (TypeOf value Is GridColumn OrElse TypeOf value Is GridControlBand)
		End Function

		Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
			Throw New NotImplementedException()
		End Function
	End Class


End Namespace

File: CS/WpfApplication48/VerticalLineHelper.cs
using DevExpress.Xpf.Grid;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Data;
using System.Windows.Media;

namespace WpfApplication48
{
    public class VerticalLineHelper
    {
        public static double GetLineThickness(DependencyObject obj)
        {
            return (double)obj.GetValue(LineThicknessProperty);
        }
        public static void SetLineThickness(DependencyObject obj, double value)
        {
            obj.SetValue(LineThicknessProperty, value);
        }
        public static Brush GetLineColor(DependencyObject obj)
        {
            return (Brush)obj.GetValue(LineColorProperty);
        }
        public static void SetLineColor(DependencyObject obj, Brush value)
        {
            obj.SetValue(LineColorProperty, value);
        }
        public static readonly DependencyProperty LineThicknessProperty =
            DependencyProperty.RegisterAttached("LineThickness", typeof(double), typeof(VerticalLineHelper), new PropertyMetadata(0.0));
        public static readonly DependencyProperty LineColorProperty =
            DependencyProperty.RegisterAttached("LineColor", typeof(Brush), typeof(VerticalLineHelper), new PropertyMetadata(Brushes.Black));
    }

    public class VerticalLineThicknessConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return new Thickness(0, 0, (double)value, 0);
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }

    public class VerticalLineVisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return (value != null) && (value is GridColumn || value is GridControlBand);
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }


}
